docs/release-procedure.md

```diff
@@ -1,8 +1,52 @@
 # Release Procedure
 
-The Cloud Provider OpenStack Release is done in sync with
-kubernetes/kubernetes. Minor versions can be released intermittently for
-critical bug fixes.
+Major versions of OpenStack Cloud Provider are done in sync with
+[kubernetes/kubernetes](https://github.com/kubernetes/kubernetes).
+Minor versions can be released intermittently for critical bug fixes.
+
+## Preparing for a release
+
+Note that while we use the terms *major* and *minor* here and below, these
+actually correspond to SemVer *minor* and *patch* versions. This is discussed
+in detail in [the Kubernetes documentation](https://github.com/kubernetes/sig-release/blob/master/release-engineering/versioning.md#kubernetes-release-versioning).
+
+### Major releases (`X.Y.0`)
+
+The following steps can be submitted as separate commits in a single PR or as
+separate PRs:
+
+1. Bump the version of the sidecar container images used in both the manifests
+   and Helm Charts.
+
+    You should pay particular attention to any major version bumps, since they
+    may require additional changes to the manifests or charts.
+
+    Example: https://github.com/kubernetes/cloud-provider-openstack/pull/3008
+
+2. Bump the version of `k8s.io/kubernetes` to the latest minor version.
+
+    ```bash
+    go get -u k8s.io/kubernetes@latest
+    ```
+
+    Note that this will frequently bring in a new Go version corresponding to
+    the latest and greatest version.
+
+    Example: https://github.com/kubernetes/cloud-provider-openstack/pull/3004
+
+3. Bump all other dependencies.
+
+    Once again, pay close attention to any major version bumps of packages. You
+    will also need to manually update the `replace` directives used to fix `go
+    list -m all` invocation, updating those that are still necessary and
+    removing any that are not.
```

Comment:
I should probably note that the version of the packages in the `replace` directive should be identical to the version of `k8s.io/kubernetes`